### ISSUE TITLE
Basic scaling method is missing (val / sqrt(len))

### ISSUE THREAD

**Issue by spitfire05** [phip1611/spectrum-analyzer#41]
`divide_by_N_sqrt` function doc states that this is the scaling `rustfft` recommends. While in fact, it implements `sqrt(val / len)` when `rustfft` doc actually recommends `val / sqrt(len)` (see: https://docs.rs/rustfft/latest/rustfft/#normalization). The description for this method is misleading.

Also, probably `A sqrt(len)` scaling method should be added, as it provides a useful case.

Consider this artificial waveform consisting of two sines with frequencies (300, 50) Hz, and amplitudes (10, 2) (using `wavegen` crate):

```rust
let sample_rate = 2560.0;
let n = 2048;

let samples = Waveform::<f32>::with_components(sample_rate, vec![sine!(300, 10), sine!(50, 2)])
    .into_iter()
    .take(n)
    .collect::<Vec<f32>>();
```

Calculated spectrum with `divide_by_N_sqrt`:
```rust
let spectrum = samples_fft_to_spectrum(
          &samples,
          sample_rate as u32,
          FrequencyLimit::All,
          Some(&divide_by_N_sqrt),
      ).unwrap();

draw("spectrum.png", "Spectrum", spectrum.data().iter().map(|(f, a)| (f.val(), a.val())))?;
```

Plot:
![spectrum_wrong](https://user-images.githubusercontent.com/5302324/176860921-854bae8d-df50-4892-8fe9-ec11cb80b679.png)

Notice the amplitudes of spectrum being respectively (1.0, 2.2)

Now we do the same, but with custom scaling:
```rust
let spectrum = samples_fft_to_spectrum(
        &samples,
        sample_rate as u32,
        FrequencyLimit::All,
        None,
    ).unwrap();

draw("spectrum.png", "Spectrum", spectrum.data().iter().map(|(f, a)| (f.val(), a.val() * (1.0 / sqrtf(n as f32)))))?;
                                                                                      // ^ scalling by 1/sqrt(len)
```

Giving following spectrum plot:
![spectrum](https://user-images.githubusercontent.com/5302324/176861517-d22305ac-321d-43df-a70e-08f43cce9960.png)

Notice the spectrum amplitudes being (45, 226), which is close enough to the amplitudes of input waveform (minus the FFT leak).

This method of scaling is common in spectrum analysis, so the approximate amplitudes of waveform's components can be directly read from the spectrum.

**Comment by phip1611** [comment]
Hi! I'll look into it in the next few days. Thanks for the issue! I think it's quite cool that people are using my library 😀  @spitfire05

**Comment by phip1611** [comment]
@spitfire05 I'm not sure what you mean by 

> Also, probably a `sqrt(len)` scaling method should be added, as it provides a useful case.

Could you please look into https://github.com/phip1611/spectrum-analyzer/pull/42 and check if this is what you'd expect and what is missing?

**Comment by spitfire05** [comment]
#42 LGTM! 

Also, where I wrote `A sqrt(len)` I should've written `val / sqrt(len)`, sorry my my mistake.

**Comment by phip1611** [comment]
Fixed in `v1.2.6` on crates.io. Thanks again for the bug report! good catch
